File: AndroidJava/Java/app/src/main/java/com/paytapsintegrationjava/adpter/PaymentDetailAdapter.java
package com.paytapsintegrationjava.adpter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.paytapsintegrationjava.R;
import com.paytapsintegrationjava.model.CommonModel;

import java.util.ArrayList;

public class PaymentDetailAdapter extends RecyclerView.Adapter<PaymentDetailAdapter.ViewHolder> {
    private static final String TAG = PaymentDetailAdapter.class.getSimpleName();
    Context context;
    ArrayList<CommonModel> arrayList;

    public PaymentDetailAdapter(Context context, ArrayList<CommonModel> arrayList) {
        this.context = context;
        this.arrayList = arrayList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int i) {
        View view = LayoutInflater.from(context).inflate(R.layout.inflate_payment_detail_list_item, parent, false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }


    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int pos) {
        CommonModel model = arrayList.get(pos);

        viewHolder.tv_title.setText(model.getTitle());
        viewHolder.tv_value.setText(model.getValue());
    }


    @Override
    public int getItemCount() {
        return arrayList.size();
    }

    /**
     * View Holder for represent the single Row views
     */
    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tv_title, tv_value;
        View layout;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            layout = itemView;
            tv_title = layout.findViewById(R.id.tv_title);
            tv_value = layout.findViewById(R.id.tv_value);
        }
    }

}

File: AndroidKotlin/Kotlin/app/src/main/java/com/paytapsintegrationkotlin/adpter/PaymentDetailAdapter.kt
package com.paytapsintegrationkotlin.adpter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView


import com.paytapsintegrationkotlin.R
import com.paytapsintegrationkotlin.model.CommonModel

import java.util.ArrayList

class PaymentDetailAdapter(
    internal var context: Context,
    internal var arrayList: ArrayList<CommonModel>
) : RecyclerView.Adapter<PaymentDetailAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ViewHolder {
        val view = LayoutInflater.from(context)
            .inflate(R.layout.inflate_payment_detail_list_item, parent, false)
        return ViewHolder(view)
    }


    override fun onBindViewHolder(viewHolder: ViewHolder, pos: Int) {
        val model = arrayList[pos]

        viewHolder.tv_title.text = model.title
        viewHolder.tv_value.text = model.value
    }


    override fun getItemCount(): Int {
        return arrayList.size
    }

    /**
     * View Holder for represent the single Row views
     */
    inner class ViewHolder(internal var layout: View) : RecyclerView.ViewHolder(layout) {
        internal var tv_title: TextView
        internal var tv_value: TextView

        init {
            tv_title = layout.findViewById(R.id.tv_title)
            tv_value = layout.findViewById(R.id.tv_value)
        }
    }

    companion object {
        private val TAG = PaymentDetailAdapter::class.java.simpleName
    }

}
